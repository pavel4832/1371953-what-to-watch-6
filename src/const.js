export const RATING_STARS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

export const COLUMN_QUANTITY = 2;

export const PRECISION_RATING = 2;

export const TIMEOUT_PREVIEW = 1000;

export const CONTENT_TYPE = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const POSTER_SIZE = {
  WIDTH: `280`,
  HEIGHT: `175`
};

export const COUNT_CARD = {
  SAME: 4,
  MAIN_PER_STEP: 8,
  ACTIVE: 0
};

export const FILTER_TYPE = {
  ALL_GENRE: `All genre`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  KIDS_FAMILY: `kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const MAX_FILTERS = 9;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player/:id`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`,
  ERROR: `/error`
};

export const EMAIL_REGEXP = /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/iu;
