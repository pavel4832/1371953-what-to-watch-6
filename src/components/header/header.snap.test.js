import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import Header from './header';
import configureStore from 'redux-mock-store';
import {AuthorizationStatus} from "../../const";

const mockStore = configureStore({});

it(`Should Header render correctly`, () => {
  const store = mockStore({
    USER: {authorizationStatus: AuthorizationStatus.AUTH}
  });
  const history = createMemoryHistory();
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <Header isReview={false} pageType={`user`} headerTitle={``} />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
