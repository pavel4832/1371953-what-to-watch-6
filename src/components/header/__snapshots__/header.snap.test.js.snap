// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly 1`] = `
<div>
  <header
    class="page-header movie-card__head user-page__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    
    <h1
      class="page-title user-page__title"
      style="display: none;"
    />
    <div
      class="user-block"
    >
      <div
        class="user-block__avatar"
      >
        <a
          href="/mylist"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </a>
      </div>
      <a
        class="user-block__link"
        href="/"
      >
        Log out
      </a>
    </div>
  </header>
</div>
`;
